Plugins/Json/src/DetectorJsonConverter.cpp

```diff
@@ -98,117 +100,164 @@ nlohmann::json Acts::DetectorJsonConverter::toJsonDetray(
+  nlohmann::json jGeometryHeader;
+  nlohmann::json jGeometryVolumes;
+  // (2) Surface grids
   nlohmann::json jSurfaceGrids;
   nlohmann::json jSurfaceGridsData;
   nlohmann::json jSurfaceGridsInfoCollection;
   nlohmann::json jSurfaceGridsHeader;
-  for (const auto [iv, volume] : enumerate(volumes)) {
-    // And its surface navigation delegates
-    nlohmann::json jSurfacesDelegate = IndexedSurfacesJsonConverter::toJson(
-        volume->internalNavigation(), true);
-    if (jSurfacesDelegate.is_null()) {
-      continue;
-    }
-    // Colplete the grid json for detray usage
-    jSurfacesDelegate["owner_link"] = iv;
-    // jSurfacesDelegate["acc_link"] =
-    nlohmann::json jSurfaceGridsCollection;
-    jSurfaceGridsCollection.push_back(jSurfacesDelegate);
-
-    nlohmann::json jSurfaceGridsInfo;
-    jSurfaceGridsInfo["volume_link"] = iv;
-    jSurfaceGridsInfo["grid_data"] = jSurfaceGridsCollection;
-    jSurfaceGridsInfoCollection.push_back(jSurfaceGridsInfo);
-  }
-  jSurfaceGridsData["grids"] = jSurfaceGridsInfoCollection;
-
-  jCommonHeader["tag"] = "surface_grids";
-  jSurfaceGridsHeader["common"] = jCommonHeader;
-  jSurfaceGridsHeader["grid_count"] = jSurfaceGridsInfoCollection.size();
-
-  jSurfaceGrids["header"] = jSurfaceGridsHeader;
-  jSurfaceGrids["data"] = jSurfaceGridsData;
-
-  jFile["surface_grids"] = jSurfaceGrids;
-
-  // (3) material section
-  jCommonHeader["tag"] = "material_maps";
-
+  // (3) Material
   nlohmann::json jMaterial;
   nlohmann::json jMaterialData;
   nlohmann::json jMaterialHeader;
-  jMaterialHeader["common"] = jCommonHeader;
-
   nlohmann::json jMaterialGrids;
+
+  // Counters:
+  std::size_t nSurfaces = 0;
   std::size_t nGrids = 0;
-  for (const auto [iv, volume] : enumerate(volumes)) {
+
+  // Main loop over volumes
+  for (auto [iv, volume] : enumerate(detector.volumes())) {
+    // Write the volume information
+    nlohmann::json jVolume;
+    jVolume["name"] = volume->name();
+
+    // Write the transform and bounds information
+    jVolume["transform"] = Transform3JsonConverter::toJson(
+        volume->transform(gctx), options.volumeOptions.transformOptions);
+    jVolume["bounds"] =
+        VolumeBoundsJsonConverter::toJson(volume->volumeBounds());
+    auto volumeBoundsType = volume->volumeBounds().type();
+    if (volumeBoundsType == VolumeBounds::BoundsType::eCylinder) {
+      jVolume["type"] = 0u;
+    } else if (volumeBoundsType == VolumeBounds::BoundsType::eCuboid) {
+      jVolume["type"] = 4u;
+    } else {
+      throw std::runtime_error("Unsupported volume bounds type");
+    }
+    // Link to itself
+    jVolume["index"] = iv;
+
+    // Acceleration link if there
+    nlohmann::json jSurfacesDelegate = IndexedSurfacesJsonConverter::toJson(
+        volume->internalNavigation(), true);
+    if (!jSurfacesDelegate.is_null()) {
+      // Colplete the grid json for detray usage
+      jSurfacesDelegate["owner_link"] = iv;
+      // jSurfacesDelegate["acc_link"] =
```

Comment:
I know this is old code, but should this line be removed at some point?

Reply:
Will do in the next round, though hopefully we will not keep this code for all too long, and replace it by native detray writing after the `Plugins/Detray` has been established.